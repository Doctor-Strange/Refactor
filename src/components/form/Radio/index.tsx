import React, { useEffect, useState, useRef } from "react";

const Radio = (props: IRadio) => {
  const [checked, setChecked] = useState(null);
  const wrapperRef = useRef(null);

  useEffect(() => {
    setChecked(props.defaultCheck);
  }, [props.defaultCheck]);

  useEffect(() => {
    if (props.error_status) {
      scrollTo(0, wrapperRef.current.offsetTop);
    }
  }, [props.error_status]);

  return props.data.map((i: any, index) => {
    return (
      <div
        className={[
          "radio_container",
          checked === i.value ? "active_radio_item" : null,
        ].join(" ")}
        key={index}
        ref={wrapperRef}
      >
        <label className="container">
          {i.label}
          {i.extra_text && <p>{i.extra_text}</p>}
          <input
          data-test-id="radio"
            type="radio"
            name={props.name}
            value={i.value}
            checked={checked === i.value}
            onChange={(e) => {
              props.SelectHandler(+e.target.value);
            }}
          />
          <span className="checkmark"></span>
        </label>
      </div>
    );
  });
};

interface IRadio {
  name: string;
  data: any;
  SelectHandler: any;
  defaultCheck?: number;
  error_status?: boolean;
}

export default Radio;
