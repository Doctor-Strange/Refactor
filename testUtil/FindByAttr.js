

export const FindByAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`);
};
