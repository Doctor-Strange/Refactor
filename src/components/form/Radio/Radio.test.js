import React from "react";
import { mount } from "enzyme";
import { FindByAttr } from "../../../../testUtil/FindByAttr";
import Radio from "./";

const SelectHandlerMock = jest.fn();

const props = {
  data: [
    { value: "a", text: "a" },
    { value: "b", text: "b" },
    { value: "c", text: "c" },
  ],
  name: "name",
  SelectHandler: SelectHandlerMock,
};
const setup = (props) => {
  return mount(<Radio {...props} />);
};

describe("Radio component", () => {
  describe("snapshot", () => {
    test("/component/form/Radio snapshot", () => {
      const wrapper = setup(props);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe("Click om radio", () => {
    test("check mocks functions", () => {
      const wrapper = setup(props);
      const radio = FindByAttr(wrapper, "radio");
      radio.first().simulate("change", { target: { value: 0 } });
      expect(SelectHandlerMock).toHaveBeenCalledTimes(1);
      expect(SelectHandlerMock).toHaveBeenCalledWith(0);
    });
  });
});
