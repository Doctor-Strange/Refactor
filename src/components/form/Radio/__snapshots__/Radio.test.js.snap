// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component snapshot /component/form/Radio snapshot 1`] = `ReactWrapper {}`;
